src/lex.rs

```diff
@@ -844,13 +844,15 @@ impl<'a> Iterator for Lexer<'a> {
         if self.debug {
             println!("lexeme: {:?}", c);
         }
-        c.map(|result| result.map_err(CompileError::Lex))
+        // oof
+        c.map(|result| result.map_err(|err| err.map(Error::GenericLex)))
```

Comment:
This will be handled later in the error mega-issue when the Generic* errors are removed. We will basically have to propagate all the string to error conversions to their sources.